adaptdl/adaptdl/torch/__init__.py

```diff
@@ -48,8 +60,18 @@ def init_process_group(backend):
                 break
         response.raise_for_status()
         master_addr = response.json()[0]
+        sched_version = os.environ.get("ADAPTDL_SCHED_VERSION")
+        trainer_version = pkg_resources.get_distribution("adaptdl").version
+        if version_check(sched_version, lib="sched") and \
+                version_check(trainer_version, lib="trainer"):
+            trainer_ver_maj = semver.VersionInfo.parse(trainer_version).major
+            sched_ver_maj = semver.VersionInfo.parse(sched_version).major
+            if trainer_ver_maj != sched_ver_maj:
+                raise Exception('The adaptdl version between trainer'
+                                ' and scheduler should be compatible')
```

Comment:
Provide more info here, e.g. "adaptdl version x.x.x is incompatible with scheduler version x.x.x"